packages/build/src/core/bin.js

```diff
@@ -30,7 +30,8 @@ const runCli = async function () {
 }
 
 const parseFlags = function () {
-  const { featureFlags: cliFeatureFlags = '', ...flags } = yargs.options(FLAGS).usage(USAGE).parse()
+  const yarg = yargs()
+  const { featureFlags: cliFeatureFlags = '', ...flags } = yarg.options(FLAGS).usage(USAGE).parse()
```

Comment:
What are your thoughts on using `hideBin()` as well, like in https://github.com/netlify/build-info/blob/36fad7ce68bacda5741f6c20a40675fde736a595/src/bin.js#L23, as suggested in https://github.com/yargs/yargs#simple-example?

Reply:
Indeed, that I didn't try because even though I saw it in the examples, it's not mentioned in the migration notes. Worth trying that.

Do you have the time to experiment with this yourself? If so, let's close this PR :)

Reply:
Did you have something like this in mind?

```js
const { featureFlags: cliFeatureFlags = '', ...flags } = yargs(hideBin(process.argv)).options(FLAGS).usage(USAGE).parse()
```

Reply:
Yes, absolutely!

Reply:
I pushed the change but I don't think it fixes anything :/

Reply:
Should we try to upgrade to v16 first, then to v17 in a separate PR?
I am trying v16, and it seems to work, except for some test snapshots which need to be updated (using `ava -u`) due to the `--help` output being wrapped slightly different.

Reply:
Yup, I'll try that. v16 is what zip-it-and-ship-it is currently using:

```
netlify-cli@8.9.2 C:\Users\XhmikosR\Desktop\cli
+-- @netlify/build@26.1.3
| `-- yargs@15.4.1
+-- @netlify/config@17.0.3
| `-- yargs@15.4.1 deduped
`-- @netlify/zip-it-and-ship-it@5.4.0
  `-- yargs@16.2.0
```

Reply:
I'm having trouble updating the snapshots though, because tests fail. Perhaps you could update them here?

Reply:
Do the tests still fail after using `yargs()` + `hideBin()` (except for the test snapshots)?

Reply:
Running ava kills my machines :/

It spawns like 400 processes and is using all RAM. So this makes it pretty hard for me to run the tests, unfortunately.

I'm pretty sure it's not my systems to blame. It could be ava + Windows, but I've yet to find a workaround.

Anyway, I updated the snapshots, but please try this manually too :)

Reply:
Would one of the following help?
  - `ava -c=1`: runs one test file at a time, instead of multiple in parallel. Tests will take longer, but will be easier on your machine's resources.
  - `ava --serial`: also runs one test at a time inside each test file. Tests will be even longer, but should be even easier on your machine.